public/javascripts/Gallery/src/modal/Modal.js

```diff
@@ -0,0 +1,174 @@
+     */
+    function populateModalDescriptionFields() {
+        // Adds the severity display to the Modal
+        new SeverityDisplay(self.severity, properties.severity, true)
+        // Adds the tag display to the Modal
+        new TagDisplay(self.tags, properties.tags, true)
+        // Adds the information about the temporary property to the Modal
+        let temporaryHeader = document.createElement('div')
+        let temporaryText = properties.temporary ? "Yes" : "No"
+        temporaryHeader.innerHTML = `<div><b>${i18next.t("temporary")}</b></div><div>${temporaryText}</div>`
+        self.temporary.append(temporaryHeader)
+        // Adds the information about the description of the label to the Modal
+        let descriptionText = properties.description === null ? "" : properties.description
+        let descriptionObject = document.createElement('div')
+        descriptionObject.innerHTML = `<div><b>${i18next.t("description")}</b></div><div>${descriptionText}</div>`
+        self.description.append(descriptionObject)
+    }
+
+    /**
+     * Performs the actions needed to open the modal
+     */
+    function openModal() {
+        resetModal()
+        populateModalDescriptionFields()
+        self.pano.setPano(properties.gsv_panorama_id, properties.heading, properties.pitch, properties.zoom)
+        self.pano.renderLabel(self.label)
+        self.header.text(i18next.t('gallery.' + properties.label_type))   
+    }
+
+    /**
+     * Updates the local variables to the properties of a new label and creates a 
+     * GalleryPanoramaLabel object that uses the new label properties
+     * 
+     * @param newProps The new properties to push into the Modal
+     */
+    function updateProperties(newProps) {
+        for (let attrName in newProps) {
+            properties[attrName] = newProps[attrName]
+        }
+        self.label = new GalleryPanoramaLabel(properties.label_id, properties.label_type, 
+                                              properties.canvas_x, properties.canvas_y, 
+                                              properties.canvas_width, properties.canvas_height, 
+                                              properties.heading, properties.pitch, properties.zoom)
+    }
+
+    /**
+     * Updates the index of the current label being displayed in the modal.
+     * 
+     * @param {Number} newIndex The new index of the card being displayed 
+     */
+    function updateCardIndex(newIndex) {
+        updateModalCardByIndex(newIndex)
+    }
+
+    /**
+     * Tries to update the current card to the given input index.
+     * 
+     * @param {Number} index The index of the card to update to
+     */
+    function updateModalCardByIndex(index) {
+        self.cardIndex = index;
+        updateProperties(sg.cardContainer.getCardByIndex(index).getProperties())
+        openModal()
+        let page = sg.cardContainer.getCurrentPage()
+        if (index > (page - 1) * 9) {
```

Comment:
These numbers should be set up as constants somewhere. Not sure what the numbers mean. Cards per page?

Reply:
Done